File: Level2/이진 변환 반복하기.py
s = "110010101001"

def solution(s):
    count = 0
    num = 0
    while s != '1':
        num += s.count('0')
        s = s.replace('0', '')
        target = format(len(s), 'b')
        count += 1
        s = str(target)
    return [count, num]


did = solution(s)
print(did)

File: Level2/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public String change(int num) {
        StringBuilder sb = new StringBuilder();
        while(num > 0) {
            sb.append(num % 2);
            num /= 2;
        }
        return sb.reverse().toString();
    }
    
    public int[] solution(String s) {
        int[] answer = new int[2];
        String k = s;
        while(!k.equals("1")) {
            int size = k.length();
            String tmp = k.replace("0", "");
            answer[1] += size - tmp.length();
            
            int num = Integer.valueOf(tmp.length());
            k = change(num);
            answer[0]++;
        }
        return answer;
    }
}
